src/Serilog/Core/Enrichers/FunctionPropertyEnricher.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013-2015 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#nullable enable
+using System;
+using Serilog.Events;
+
+namespace Serilog.Core.Enrichers
+{
+    /// <summary>
+    /// Adds a new property enricher to the log event.
+    /// </summary>
+    public class FunctionPropertyEnricher<T> : ILogEventEnricher
+    {
+        readonly string _name;
+        readonly Func<T> _value;
+        readonly bool _destructureObjects;
+
+        /// <summary>
+        /// Create a new property enricher.
+        /// </summary>
+        /// <param name="name">The name of the property.</param>
+        /// <param name="value">The delegate to the property.</param>
+        /// <returns>A handle to later remove the property from the context.</returns>
```

Comment:
```suggestion
```

Reply:
Interesting you flagged this to remove as I took the PropertyEnricher.cs as the template to do this and that line is also in there.  However this isn't getting merged anyway but you might want to double check this line to make sure it's right.   https://github.com/serilog/serilog/blob/81e9dee4aaa86538f9e754fa9887a8603800274a/src/Serilog/Core/Enrichers/PropertyEnricher.cs#L35